### PR TITLE
feat: Adding an upgrade page for Access control

### LINKED ISSUE #18555: [Task]: Add an upgrade page for Access control for introducing the new feature to community users
### Is there an existing issue for this?

- [X] I have searched the existing issues

### SubTasks

_No response_

### PR DESCRIPTION
**Pull request by ankitakinger**
## Description

> Adding an upgrade page for Access control to help users know about the enterprise feature.

Fixes #18555 

## Type of change

- New feature (non-breaking change which adds functionality)

- Manual

### Dev activity
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] PR is being merged under a feature flag

### QA activity:
- [ ] Test plan has been approved by relevant developers
- [ ] Test plan has been peer reviewed by QA
- [ ] Cypress test cases have been added and approved by either SDET or manual QA
- [ ] Organized project review call with relevant stakeholders after Round 1/2 of QA
- [ ] Added Test Plan Approved label after reveiwing all Cypress test

### CODE REVIEW

File: app/client/src/ce/pages/Upgrade/Carousel.tsx [hunk 1]
```diff
@@ -9,7 +9,7 @@ const CarouselContainer = styled.div`
   gap: 64px;
   justify-content: center;
   align-items: center;
-  padding: 16px;
+  padding: 16px 52px;
```
Comment (berzerkeer) [inline app/client/src/ce/pages/Upgrade/Carousel.tsx:12]: Should we also get anubhav's review here since these css changes are made in the source itself.

**Review by berzerkeer — COMMENTED** [review]
@ankitakinger

**Review by IAmAnubhavSaini — COMMENTED** [review]
Please verify that earlier audit logs upgrade page doesn't change visually. I think it will; then either get approval from design team for audit-logs or make changes in such a way that upgrade page structure remains but audit-logs and access-control can send custom style into UpgradePage component, via props or class-names.

Basically, it should not affect the earlier visual design.

**Comment by ankitakinger** [comment]
@IAmAnubhavSaini @berzerkeer 

Before:
https://www.loom.com/share/2c6cf6b78e6b4ec58d39fd989bb46300

After:
https://www.loom.com/share/eb6bad159b4f45c2972de06817d0dfe3

@vinay-appsmith @Debsourabh @vasanth-appsmith what would be your inputs on the same? Let me know if anything is breaking as per the designs due to my changes, or if I should change something here.

**Comment by IAmAnubhavSaini** [comment]
This is how it looked on my screen earlier

Screenshot 2022-12-01 at 10 25 30 AM

and this is how it looks now (on this branch)

Screenshot 2022-12-01 at 10 34 38 AM


The right side images were supposed to be pushed to the right as per design, as that was purely a design requirement.

**Comment by IAmAnubhavSaini** [comment]
The images have weird borders around them? Are these screenshots instead of exports?

Screenshot 2022-12-01 at 10 39 04 AM

Screenshot 2022-12-01 at 10 38 42 AM

**Comment by IAmAnubhavSaini** [comment]
Same before and after for different screen configuration

Before

Screenshot 2022-12-01 at 10 40 11 AM

After

Screenshot 2022-12-01 at 10 38 29 AM

**Comment by ankitakinger** [comment]
@IAmAnubhavSaini just pushed a change for fixing the UI on different screen sizes.
Regarding images having borders, these images will be getting changed later and we are pushing Access control upgrade page under feature flag right now.

**Comment by IAmAnubhavSaini** [comment]
Please merge with correct images. Highlight any blocker in getting the final images.

**Comment by ankitakinger** [comment]
@IAmAnubhavSaini This was asked by @trishaanand to be merged now with these images and update later as this is behind feature flag anyway.

**Comment by IAmAnubhavSaini** [comment]
@trishaanand What is the value we are deriving from it merging today without correct images?

**Comment by trishaanand** [comment]
> @trishaanand What is the value we are deriving from it merging today without correct images?

@IAmAnubhavSaini : I was of the opinion that as a tech task this can be completed and if there are any changes to the images or content, this can be picked up by anyone at all in the team to do the updates. As this does not impact any user (since its behind feature flag), there is no harm as well.

**Review by trishaanand — APPROVED** [review]

**Pull request merged**